csi/controller.go

```diff
@@ -980,6 +1016,88 @@ func (s *OsdCsiServer) CreateSnapshot(
 		},
 	}, nil
 }
+func (s *OsdCsiServer) getCloudBackupClient(ctx context.Context) (api.OpenStorageCloudBackupClient, error) {
+	// Get grpc connection
+	conn, err := s.getRemoteConn(ctx)
+	if err != nil {
+		return nil, status.Errorf(
+			codes.Unavailable,
+			"Unable to connect to SDK server: %v", err)
+	}
+	return s.cloudBackupClient(conn), nil
+}
+
+func (s *OsdCsiServer) createCloudBackup(
+	ctx context.Context,
+	req *csi.CreateSnapshotRequest,
+) (*csi.CreateSnapshotResponse, error) {
+	cloudBackupClient, err := s.getCloudBackupClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get any labels passed in by the CO
+	_, locator, _, err := s.specHandler.SpecFromOpts(req.GetParameters())
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Unable to get parameters: %v", err)
+	}
+
+	backupID := req.GetName()
+	// Create snapshot
+	_, err = cloudBackupClient.Create(ctx, &api.SdkCloudBackupCreateRequest{
+		VolumeId: req.GetSourceVolumeId(),
```

Comment:
The CloudBackupCreate request also takes in "CredentialID". We should take that as an input from the user.  Similar to `snapshot-type`, lets take an input `credential-id`